src/tribler/core/components/libtorrent/restapi/downloads_endpoint.py

```diff
@@ -465,7 +466,7 @@ async def update_download(self, request):
             num_files = len(download.tdef.get_files())
             if not all([0 <= index < num_files for index in selected_files_list]):
                 return RESTResponse({"error": "index out of range"}, status=HTTP_BAD_REQUEST)
-            download.set_selected_files(selected_files_list)
+            download.set_selected_files(selected_files_list, no_tree=True)
```

Comment:
This way, we leave the `file_instance.selected` attribute for `tdef.torrent_file_tree` items not updated; it may be OK to do this, but I think at least we should note this in a comment, so further we should not rely on values of tree item `.selected` attribute.